src/pages/instances/forms/DuplicateInstanceForm.tsx

```diff
@@ -0,0 +1,290 @@
+    useQuery({
+      queryKey: [queryKeys.cluster],
+      queryFn: fetchClusterMembers,
+      enabled: isClustered,
+    });
+
+  const { data: instances = [] } = useQuery({
+    queryKey: [queryKeys.instances],
+    queryFn: () => fetchInstances(instance.project),
+  });
+
+  const notifySuccess = (instanceName: string, instanceProject: string) => {
+    const message = (
+      <>
+        Created Instance <strong>{instanceName}</strong>.
+      </>
+    );
+
+    const actions = [
+      {
+        label: `${instanceName}`,
+        onClick: () =>
+          navigate(`/ui/project/${instanceProject}/instance/${instanceName}`),
+      },
+    ];
+
+    toastNotify.success(message, actions);
+  };
+
+  const getDuplicatedInstanceName = (instance: LxdInstance): string => {
+    const instanceNames = instances.map((instance) => instance.name);
+    const newInstanceName = truncateInstanceName(instance.name, "-duplicate");
+
+    if (instanceNames.includes(newInstanceName)) {
+      let count = 1;
+      while (instanceNames.includes(`${newInstanceName}-${count}`)) {
+        count++;
+      }
+      return `${newInstanceName}-${count}`;
+    }
+    void queryClient.invalidateQueries({
+      predicate: (query) => {
+        return query.queryKey[0] === queryKeys.instances;
+      },
+    });
+
+    return newInstanceName;
+  };
+
+  const formik = useFormik<LxdInstanceDuplicate>({
+    initialValues: {
+      instanceName: getDuplicatedInstanceName(instance),
+      targetProject: instance.project,
+      allowInconsistent: false,
+      instanceOnly: false,
+      stateful: false,
+      targetClusterMember: isClustered ? instance.location : "",
+      targetStoragePool:
+        (instance.devices["root"] as LxdDiskDevice)?.pool ??
+        storagePools[0]?.name,
+    },
+    enableReinitialize: true,
+    validationSchema: Yup.object().shape({
+      instanceName: instanceNameValidation(
+        instance.project,
+        controllerState,
+      ).required(),
+    }),
+    onSubmit: (values) => {
+      createInstance(
+        JSON.stringify({
+          description: instance.description,
+          name: values.instanceName,
+          architecture: instance.architecture,
+          source: {
+            allow_inconsistent: values.allowInconsistent,
+            instance_only: values.instanceOnly,
+            source: instance.name,
+            type: "copy",
+            project: instance.project,
+          },
+          stateful: values.stateful,
+          devices: {
+            ...instance.devices,
+            root: {
+              path: "/",
+              type: "disk",
+              pool: values.targetStoragePool,
+            },
+          },
+        }),
+        values.targetProject,
+        values.targetClusterMember,
+      )
+        .then((operation) => {
+          toastNotify.info(`Duplication of instance ${instance.name} started.`);
+          close();
```

Comment:
there is a call to `close()` in both `then` and `catch` block. In this case, you can call `close()` once inside a `finally` block